### ISSUE TITLE
Resource builder with_attributes doesn't work?

### ISSUE THREAD

**Issue by akantsevoi** [open-telemetry/opentelemetry-rust#3034]
### What happened?

When I set up meter like this

```rust
pub fn init_meter_provider(endpoint: String) -> Result<SdkMeterProvider, Box<dyn std::error::Error>> {
  let exporter = MetricExporter::builder()
    .with_tonic()
    .with_endpoint(endpoint)
    .with_temporality(Temporality::Cumulative)
    .build()
    .expect("exp");

  let resource = Resource::builder_empty()
    .with_attribute(KeyValue::new("peer_id", "some_string_to_send"))
    .with_service_name("some_service_name")
    .build();

  let provider = SdkMeterProvider::builder()
      .with_periodic_exporter(exporter)
      .with_resource(resource)
      .build();

  global::set_meter_provider(provider.clone());
  Ok(provider)
}
```

 I'm creating Resource as well and try to provide some attribute that will be reported in all counter, I expect it to work similar to what we have in go: https://opentelemetry.io/docs/languages/go/resources/ but for some reason these attributes are not reported. What am I doing wrong?

### OpenTelemetry API Version (i.e version of `opentelemetry` crate)

0.30.0

### OpenTelemetry SDK Version (i.e version of `opentelemetry_sdk` crate)

0.30.0

### What Exporter(s) are you seeing the problem on?

OTLP

### Relevant log output

```shell

```

**Comment by cijothomas** [comment]
> but for some reason these attributes are not reported

Can you share more? If using OTLP Exporter, please share the output from OTel Collector's `debug` exporter, as shown in the example https://github.com/open-telemetry/opentelemetry-rust/tree/main/opentelemetry-otlp/examples/basic-otlp#metric

**Comment by akantsevoi** [comment]
Sure
```
2025-06-21T07:19:47.747Z	info	ResourceMetrics #0
Resource SchemaURL: 
Resource attributes:
     -> service.name: Str(some_service_name)
     -> peer_id: Str(some_string_to_send)
ScopeMetrics #0
ScopeMetrics SchemaURL: 
InstrumentationScope etcd_epoch_coordinator 
Metric #0
Descriptor:
     -> Name: etcd_requests
     -> Description: 
     -> Unit: 
     -> DataType: Sum
     -> IsMonotonic: true
     -> AggregationTemporality: Cumulative
NumberDataPoints #0
Data point attributes:
     -> success: Bool(false)
StartTimestamp: 2025-06-21 07:19:37.577029807 +0000 UTC
Timestamp: 2025-06-21 07:19:47.561576587 +0000 UTC
Value: 4
	{"resource": {"service.instance.id": "09cc295d-0ee2-405e-8ba1-0cd1bceaf534", "service.name": "otelcol-contrib", "service.version": "0.128.0"}, "otelcol.component.id": "debug", "otelcol.component.kind": "exporter", "otelcol.signal": "metrics"}
2025-06-21T07:20:15.396Z	info	Metrics	{"resource": {"service.instance.id": "09cc295d-0ee2-405e-8ba1-0cd1bceaf534", "service.name": "otelcol-contrib", "service.version": "0.128.0"}, "otelcol.component.id": "debug", "otelcol.component.kind": "exporter", "otelcol.signal": "metrics", "resource metrics": 1, "metrics": 2, "data points": 3}

```

If I understand these things well:
```
Resource attributes:
     -> service.name: Str(some_service_name)
     -> peer_id: Str(some_string_to_send)
```

Resource attirbutes are correctly send to otel, so the error not in SDK but somewhere between otel and prometheus?
